stacks_queues/queue.py

```diff
@@ -23,39 +23,67 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException("Queue is full!")
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException("Queue is empty!")
+    
+        front_ele = self.store[self.front]
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+        return front_ele
+
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        if not self.store:
+            return None
+        return self.store[0]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.store.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size == 0:
+            return True
+        else:
+            return False
```

Comment:
Try to avoid the anti-pattern of testing a condition, and then returning the value that's the same results as the condition. Prefer
```suggestion
        return self.size == 0
```